scene/gui/control.cpp

```diff
@@ -1845,19 +1845,46 @@ Control::FocusMode Control::get_focus_mode() const {
 	return data.focus_mode;
 }
 
+void Control::set_focus_layer(int p_layer_id) {
+	// If this Control has focus, then lets find the next valid Control to focus before switching layers.
+	if (has_focus()) {
+
+		Control *next = find_next_valid_focus();
+
+		if (next) {
+			next->grab_focus(false);
+			get_viewport()->set_input_as_handled();
```

Comment:
What is this for?

Reply:
Great question. It was probably a copy/paste thing from... somewhere. Either way, I removed it and tested, works fine without it.